modules/apps/object/object-rest-api/src/main/java/com/liferay/object/rest/dto/v1_0/ObjectEntry.java

```diff
@@ -397,7 +397,33 @@ public void setKeywords(
 	@JsonAnyGetter
 	@Schema
 	@Valid
+	public Map<String, UnsafeSupplier<Object, Exception>> getLazyProperties() {
+		return _lazyProperties;
+	}
+
 	public Map<String, Object> getProperties() {
+		if (properties != null) {
+			return properties;
+		}
+
+		if (_lazyProperties != null) {
+			properties = new HashMap<>();
+
+			_lazyProperties.forEach(
+				(key, value) -> {
+					try {
+						properties.put(key, value.get());
+					}
+					catch (Throwable e) {
+						throw new RuntimeException(e);
+					}
+				});
+
+			_propertiesSupplier = null;
```

Comment:
I think with this use case, we can get rid of the `_propertiesSupplier` parameter